<launch>
    <!-- Launch the STDR simulator with map and GUI -->
    <include file="$(find stdr_launchers)/launch/server_with_map_and_gui_plus_robot.launch" />

    <!-- Define an argument to control if the dead reckoning node should run -->
    <arg name="run_steering" default="true" />

    <!-- Launch rqt_robot_steering if the argument is false -->
    <group if="$(arg run_steering)">
        <!-- Execute rosrun for rqt_robot_steering -->
        <node pkg="rqt_robot_steering" type="rqt_robot_steering" name="rqt_robot_steering" output="screen"/>
    </group>

    <!-- Load Parameters from YAML -->
    <rosparam file="$(find robot_no_crash)/config/safety.yaml" command="load" ns="robot_no_crash_node"/>

    <!-- Launch the NoCrash Node -->
    <node pkg="robot_no_crash" type="robot_no_crash.py" name="robot_no_crash_node" output="screen">
        <remap from="cmd_vel" to="/robot0/cmd_vel"/>
    </node>
</launch>

